core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -261,7 +257,9 @@ class TransactionMarkerChannelManager(
     }.filter { case (_, entries) => !entries.isEmpty }.map { case (node, entries) =>
       val markersToSend = entries.asScala.map(_.txnMarkerEntry).asJava
       val requestCompletionHandler = new TransactionMarkerRequestCompletionHandler(node.id, txnStateManager, this, entries)
-      val request = new WriteTxnMarkersRequest.Builder(writeTxnMarkersRequestVersion, markersToSend)
+      val request = new WriteTxnMarkersRequest.Builder(
+        config.interBrokerProtocolVersion.writeTxnMarkersRequestVersion(), markersToSend
```

Comment:
It seems that we should get the MV from `metadataCache.metadataVersion() `instead of `config.interBrokerProtocolVersion` since MV can be set dynamically. @jolshan : What do you think?

Reply:
Yes. Config.interbrokerProtocolVersion is not the reliable way to get MV.

Reply:
`config.interBrokerProtocolVersion` is used in code base. in zk mode, `metadataCache.metadataVersion` is static config. By contrast, kraft has dynamic MV. Hence, we should apply that change (`config.interBrokerProtocolVersion` -> `metadataCache.metadataVersion`) to code base. BTW, that will be a large PR I guess @junrao @jolshan WDYT?

Reply:
Taking a quick look, it doesn't look like it is used in too many places, but can definitely be done as a followup.

Reply:
> but can definitely be done as a followup.

@m1a2st WDYT? Sorry that I'm in traveling so can't check the code usages

Reply:
Yes, If we want to change `config.interBrokerProtocolVersion -> metadataCache.metadataVersion`, there are my finding:
1. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/cluster/Partition.scala#L143
2. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala#L1799
3. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L458
4. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L1802
5. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L1832
6. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L1901
7. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L2532
8. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaApis.scala#L2533
9. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ConfigHandler.scala#L174
10. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ConfigHandler.scala#L169
@chia7712, PTAL, Thanks you

Reply:
@m1a2st thanks for sharing. It seems we can address it in this PR as it is not used in too many places.